src/onnx/parse_matmul.cpp

```diff
@@ -148,11 +224,124 @@ struct parse_matmul : op_parser<parse_matmul>
+                                         instruction_ref& scale,
+                                         instruction_ref& zp,
+                                         bool no_zp)
+    {
+        if(no_zp)
+        {
+            scale = info.add_instruction(make_op("transpose", {{"permutation", {0, 1}}}), scale);
+        }
+        else
+        {
+            scale = info.add_instruction(make_op("transpose", {{"permutation", {0, 1}}}), scale);
+            zp    = info.add_instruction(make_op("transpose", {{"permutation", {1, 0}}}), zp);
+        }
+    }
+
+    static instruction_ref handle_dequantized(const onnx_parser::node_info& info,
+                                              const instruction_ref& a0,
+                                              const instruction_ref& scale_a0,
+                                              const instruction_ref& zp_a0,
+                                              bool no_zp)
+    {
+        instruction_ref dequantized_op;
+
+        if(no_zp)
+        {
+            auto bc_scale_a0 = info.add_instruction(
+                make_op("multibroadcast", {{"out_lens", a0->get_shape().lens()}}), scale_a0);
+            dequantized_op = info.add_instruction(make_op("dequantizelinear"), a0, bc_scale_a0);
+        }
+        else
+        {
+            auto bc_scale_a0 = info.add_instruction(
+                make_op("multibroadcast", {{"out_lens", a0->get_shape().lens()}}), scale_a0);
+
+            auto bc_zp_a0 = info.add_instruction(
+                make_op("multibroadcast", {{"out_lens", a0->get_shape().lens()}}), zp_a0);
+
+            dequantized_op =
+                info.add_instruction(make_op("dequantizelinear"), a0, bc_scale_a0, bc_zp_a0);
+        }
+        return dequantized_op;
+    }
+
+    static instruction_ref handle_scaled_output(const onnx_parser::node_info& info,
+                                                const instruction_ref& a0,
+                                                const instruction_ref& a1,
+                                                const instruction_ref& scale_a0,
+                                                const instruction_ref& scale_a1,
+                                                const instruction_ref& zp_a0,
+                                                const instruction_ref& zp_a1,
+                                                const instruction_ref& scaled_bias,
+                                                const bool has_scale_bias)
+    {
+
+        instruction_ref unsq_zp_a0;
+        instruction_ref unsq_zp_a1;
+
+        bool a0_has_no_zp = (a0 == zp_a0);
+        bool a1_has_no_zp = (a1 == zp_a1);
+
+        if(not a0_has_no_zp)
+        {
+            unsq_zp_a0 = info.add_instruction(make_op("unsqueeze", {{"axes", {0}}}), zp_a0);
+            if(zp_a0->get_shape().scalar())
+            {
+                unsq_zp_a0 =
+                    info.add_instruction(make_op("unsqueeze", {{"axes", {0}}}), unsq_zp_a0);
```

Comment:
Minor: Can these unsqueeze operators be merged to either be `unsqueeze {{axes, 0}}` and `unsqueeze {{axes, 0, 1}`?

Reply:
I dont think so here as one is assume the input is scalar and needs the additional added 1 dimension